nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -528,10 +529,12 @@ in
 
             pkgs = mkOption {
               type = types.nullOr types.attrs;
-              default = null;
-              example = literalExample "pkgs";
+              default = pkgs;
```

Comment:
Any reason you changed the default here? Isn't `null` the better default as this means that by default, overlays in existing configs from 20.03 still apply?

Reply:
To quote myself:
> The default behavior (including its performance benefits by re-using pkgs) is unchanged.

You may have misread the this change, for the reason that the current version of `nixos-containers.nix` is a bit flawed:
The values `null` and `pkgs` for `containers.pkgs` are equivalent; the extra allowed value `null` is completely redundant. It is with this PR that `null` unlocks special behvior (namely container-configured pkgs).